dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -548,17 +697,131 @@ public void test_MultilangContent_IndexingAllLanguages()
+                    Assert.assertEquals("Content in default Language gives NO results", 0,
+                            searchResults.getTotalResults());
+                }
+
+            }
+
+            if(testCase.expectedResultsWhenSearchingContentInSecondLanguage !=null) {
+                SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "catalina", 0, 10);
+                if(testCase.expectedResultsWhenSearchingContentInSecondLanguage.pageInDefaultLang) {
+                    Assert.assertTrue("Content in second Language gives results",
+                            searchResults.getTotalResults() >= 1);
+                    Assert.assertEquals(pageDefaultLang.getTitle(),
+                            searchResults.getResults().get(0).getTitle());
+                    Assert.assertEquals(pageDefaultLang.getLanguageId(),
+                            searchResults.getResults().get(0).getLanguage());
+                }
+
+                if(testCase.expectedResultsWhenSearchingContentInSecondLanguage.pageInSecondLang) {
+                    Assert.assertTrue("Content in second Language gives results",
+                            searchResults.getTotalResults() >= 1);
+                    Assert.assertEquals(pageSecondLang.getTitle(),
+                            searchResults.getResults().get(0).getTitle());
+                    Assert.assertEquals(pageSecondLang.getLanguageId(),
+                            searchResults.getResults().get(0).getLanguage());
+                }
+
+                if(!testCase.expectedResultsWhenSearchingContentInSecondLanguage.pageInDefaultLang
+                        && !testCase.expectedResultsWhenSearchingContentInSecondLanguage.pageInSecondLang) {
+                    Assert.assertEquals("Content in second Language gives NO results", 0,
+                            searchResults.getTotalResults());
+                }
+            }
 
         } finally {
             Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE",
                     defaultContentToDefaultLangOriginalValue);
+
+            Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE",
+                    defaultPagetoDefaultLangOriginalValue);
         }
     }
 
+    private HTMLPageAsset createNewVersionHtmlPage(HTMLPageAsset pageDefaultLang, Language secondLang)
+            throws DotSecurityException, DotDataException {
+        Contentlet pageSecondLang = contentletAPI.find(pageDefaultLang.getInode(), systemUser, false);
+        pageSecondLang.setInode("");
+        pageSecondLang.setLanguageId(secondLang.getId());
+        pageSecondLang = contentletAPI.checkin(pageSecondLang, systemUser, false);
+        ContentletDataGen.publish(pageSecondLang);
+        return APILocator.getHTMLPageAssetAPI().fromContentlet(pageSecondLang);
+    }
+
+    private void createNewVersionAndPublishExistingEmployeeContent(Language secondLang,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'secondLang' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)